src/services/ui/src/features/submission/app-k/index.tsx

```diff
@@ -41,14 +44,28 @@ export const AppKSubmissionForm = () => {
         user,
       },
       {
-        onSuccess: () => {
+        onSuccess: async () => {
           alert.setContent({
             header: "Package submitted",
             body: "The 1915(c) Appendix K Amendment Request has been submitted.",
           });
           alert.setBannerShow(true);
           alert.setBannerDisplayOn("/dashboard");
-          nav(originPath ? { path: originPath } : { path: "/dashboard" });
+          setIsDataPolling(true);
+          await seaStatusPoller(
+            `${draft.state}-${draft.parentWaiver}`,
+            (checks) =>
+              checks.authorityIs([Authority["1915c"]]) &&
+              checks.actionIs("Amend"),
+          ).startPollingData();
+          setIsDataPolling(false);
```

Comment:
re: hookifying this, i think this is your hook's `useCallback`, and you should return the callback so this can be a one line trigger to await polling

Reply:
this also makes it an easy plug in for ActionForm and the submission rollup into the same model.

Reply:
disregard as the extra state requirement is outgoing with form refactors!